magefiles/rulesengine/repos/common.go

```diff
@@ -536,13 +536,27 @@ func SetupMultiPlatformTests() error {
 
 func SetEnvVarsForComponentImageDeployment(rctx *rulesengine.RuleCtx) error {
 	componentImage := os.Getenv("COMPONENT_IMAGE")
-	sp := strings.Split(componentImage, "@")
-	if len(sp) != 2 {
+
+	tag := rctx.ComponentImageTag
+
+	imageData := strings.SplitN(componentImage, "@", 2)
+	if len(imageData) != 2 {
 		return fmt.Errorf("component image ref expected in format 'quay.io/<org>/<repository>@sha256:<sha256-value>', got %s", componentImage)
 	}
-	os.Setenv(fmt.Sprintf("%s_IMAGE_REPO", rctx.ComponentEnvVarPrefix), sp[0])
-	os.Setenv(fmt.Sprintf("%s_IMAGE_TAG", rctx.ComponentEnvVarPrefix), rctx.ComponentImageTag)
+	repository := imageData[0]
+
+	if tag == "" {
+		if repoParts := strings.SplitN(repository, ":", 2); len(repoParts) == 2 {
```

Comment:
I'm not sure I follow the logic on this line, if I imagine having env var set to 
```
COMPONENT_IMAGE=quay.io/redhat-user-workloads/rhtap-build-tenant/build-service/build-service@sha256:20ca83111840a28c870750d576a4e2bdf0c8452f2ed3e1a21a64b9f2cbba580d
```
(taken from the description of this PR)

Reply:
will go to the [else](https://github.com/konflux-ci/e2e-tests/blob/86a6a37b47f5cad3a84a217360e48f8e77fcb84c/magefiles/rulesengine/repos/common.go#L552) and will have the same logic we have now

Reply:
Yeah, but what image ref would actually meet the `if` condition of this line?

Do you have an example of such image ref?

Reply:
@psturc i update the code with a comment